src/lib/OAuth2ServerModel.js

```diff
@@ -14,51 +14,52 @@
 *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 *    You can download the source here: https://github.com/spark/spark-server
+*
+* @flow
+*
 */
 
-var PasswordHasher = require('./PasswordHasher');
-var roles = require('./RolesController.js');
-var when = require('when');
-
-
-var OAuth2ServerModel = function (options) {
-	this.options = options;
-};
-OAuth2ServerModel.prototype = {
-
-	getAccessToken: function (bearerToken, callback) {
-		var token = roles.getTokenInfoByToken(bearerToken);
-		callback(null, token);
-	},
+const roles = require('./RolesController.js');
 
-	getClient: function (clientId, clientSecret, callback) {
-		return callback(null, { client_id: clientId });
-	},
+type Callback = (error: ?Error, result: mixed) => void;
+type GrantType = 'password';
 
-	grantTypeAllowed: function (clientId, grantType, callback) {
-		return callback(null, 'password' === grantType);
-	},
+class OAuth2ServerModel {
+  static getAccessToken(bearerToken: string, callback: Callback) {
+    const token = roles.getTokenInfoByToken(bearerToken);
+    callback(null, token);
+  }
 
-	saveAccessToken: function (accessToken, clientId, userId, expires, callback) {
-		when(roles.addAccessToken(accessToken, clientId, userId, expires))
-			.ensure(callback);
-	},
+  static getClient(clientId: string, clientSecret: string, callback: Callback) {
+    return callback(null, { client_id: clientId });
+  }
 
-	getUser: function (username, password, callback) {
-		if (username && username.toLowerCase) {
-			username = username.toLowerCase();
-		}
+  static grantTypeAllowed(clientId: string, grantType: GrantType, callback: Callback) {
+    return callback(null, grantType === 'password');
+  }
 
-		when(roles.validateLogin(username, password))
-			.then(
-			function (user) {
-				callback(null, { id: user._id });
-			},
-			function (err) {
-				callback(err, null);
-			});
-	}
-};
+  static async saveAccessToken(
+    accessToken: string,
+    clientId: string,
+    userId: string,
+    expires: number,
+    callback: Callback,
+  ) {
+    try {
+      await roles.addAccessToken(accessToken, clientId, userId, expires);
```

Comment:
Sweet :)

This code will be so much better when we migrate everything to use that.